creator-node/src/services/sync/primarySyncFromSecondary.js

```diff
@@ -84,13 +84,33 @@ async function _primarySyncFromSecondary({
     })
     decisionTree.recordStage({ name: 'getUserReplicaSet() success', log: true })
 
-    // Error if this node is not primary for user
+    // Abort if this node is not primary for user
     if (userReplicaSet.primary !== selfEndpoint) {
       decisionTree.recordStage({
-        name: 'Error - Node is not primary for user',
+        name: 'Abort - Node is not primary for user',
         data: { userReplicaSet }
       })
-      throw new Error(`Node is not primary for user`)
+
+      return {
+        abort: 'Node is not primary for user',
+        result: 'abort_current_node_is_not_user_primary'
+      }
+    }
+
+    // Also abort if the input secondary is not part of the user's secondaries
+    if (
+      userReplicaSet.secondary1 !== secondary ||
+      userReplicaSet.secondary2 !== secondary
+    ) {
+      decisionTree.recordStage({
+        name: 'Abort - Node is not secondary for user',
+        data: { userReplicaSet, requestedSecondary: secondary }
+      })
+
+      return {
+        abort: 'Node is not secondary for user',
+        result: 'abort_current_node_is_not_user_secondary'
+      }
```

Comment:
if i'm understanding this logic correctly, we should also check the secondary

Reply:
yeah good callout - but this check actually should not be here since primarySyncfromSecondary is used to sync against orphaned nodes as well i.e. non-secondary nodes

can do more intelligent checks but prob would just remove since we don't care if node is secondary or not

Reply:
will add comment about this scenario and remove code addition